src/impl/locale.js

```diff
@@ -73,7 +81,18 @@ function getCachedWeekInfo(locString) {
   return data;
 }
 
-function parseLocaleString(localeStr) {
+const localeStringCache = {};
+function getParsedCachedLocaleString(localeStr) {
```

Comment:
I can believe that this parsing is slow, but I'd like to see the benchmarks on this too